tools/accuracy_checker/accuracy_checker/data_analyzer/reidentification_analyzer.py

```diff
@@ -0,0 +1,104 @@
+from ..logging import print_info
+
+
+class ReIdentificationClassificationDataAnalyzer(BaseDataAnalyzer):
+    __provider__ = 'ReIdentificationClassificationAnnotation'
+
+    def analyze(self, result: list, meta, count_objects=True):
+        data_analysis = {}
+        positive_pairs = 0
+        negative_pairs = 0
+        for data in result:
+            positive_pairs += len(data.positive_pairs)
+            negative_pairs += len(data.negative_pairs)
+        all_pairs = positive_pairs + negative_pairs
+
+        if count_objects:
+            data_analysis['annotations_size'] = self.object_count(result)
+
+        print_info('Total pairs: {}'.format(all_pairs))
+        data_analysis['all_pairs'] = all_pairs
+        print_info('Positive pairs: {}'.format(positive_pairs))
+        data_analysis['positive_pairs'] = positive_pairs
+        print_info('Negative pairs: {}'.format(negative_pairs))
+        data_analysis['negative_pairs'] = negative_pairs
+
+        return data_analysis
+
+
+class ReIdentificationDataAnalyzer(BaseDataAnalyzer):
+    __provider__ = 'ReIdentificationAnnotation'
+
+    def analyze(self, result: list, meta, count_objects=True):
+        data_analysis = {}
+        person_in_query = Counter()
+        person_in_gallery = Counter()
+        camera_in_query = Counter()
+        camera_in_gallery = Counter()
+        unique_person = set()
+        unique_camera = set()
+
+        gallery_count = 0
+        query_count = 0
+
+        for data in result:
+            if data.query:
+                query_count += 1
+                person_in_query.update([data.person_id])
+                camera_in_query.update([data.camera_id])
+            else:
+                gallery_count += 1
+                person_in_gallery.update([data.person_id])
+                camera_in_gallery.update([data.camera_id])
+            unique_person.add(data.person_id)
+            unique_camera.add(data.camera_id)
+
+        if count_objects:
+            data_analysis['annotations_size'] = self.object_count(result)
+
+        print_info('Number of unique objects: {}'.format(len(unique_person)))
+        data_analysis['unique_objects'] = len(unique_person)
+        print_info('Count for each unique object in query:')
+        for key in person_in_query.keys():
+            print_info('{key}: {count}'.format(key=key, count=person_in_query[key]))
+        data_analysis['unique_objects_query'] = dict(person_in_query)
+        print_info('Count for each unique object in gallery:')
+        for key in person_in_gallery.keys():
+            print_info('{key}: {count}'.format(key=key, count=person_in_gallery[key]))
+        data_analysis['unique_objects_gallery'] = dict(person_in_gallery)
+
+        print_info('Number of unique cameras: {}'.format(len(unique_camera)))
+        data_analysis['unique_cameras'] = len(unique_camera)
+        print_info('Count for each unique camera in query:')
+        for key in camera_in_query.keys():
```

Comment:
why not items() here?

Reply:
or common_elements https://docs.python.org/2/library/collections.html#collections.Counter.most_common